### PR TITLE
Ports: Make git able to (mostly) clone HTTPS repositories

### PR DESCRIPTION
**Pull request by gunnarbeutner**
With these two patches applied I can _mostly_ clone Git repositories via HTTPS:

```
courage:~ $ git clone -c http.sslVerify=false https://github.com/SerenityOS/serenity
Cloning into 'serenity'...
remote: Enumerating objects: 178826, done.
remote: Counting objects: 100% (1880/1880), done.
remote: Compressing objects: 100% (907/907), done.
error: RPC failed; curl 56 OpenSSL SSL_read: error:1408F119:SSL routines:SSL3_GET_RECORD:decryption failed or bad record mac, errno 0
error: 1918 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
```

And another attempt with a smaller repository:

```
courage:~ $ git clone -v -c http.sslVerify=false https://github.com/gunnarbeutner/shop-app
Cloning into 'shop-app'...
POST git-upload-pack (164 bytes)
POST git-upload-pack (252 bytes)
remote: Enumerating objects: 820, done.
remote: Total 820 (delta 0), reused 0 (delta 0), pack-reused 820
Receiving objects: 100% (820/820), 479.24 KiB | 59. 0 KiB/s, done.
Resolving deltas: 100% (559/559), done.
courage:~ $ ls shop-app/
COPYING README.md app composer.json config.example.php doc jabber menus public schema scripts
courage:~ $ 
```

The download is excruciatingly slow, but at least it works - sometimes.

### CODE REVIEW

**Comment by alimpfard** [comment]
> BadRecordMAC

I'm glad this cursed error plagues everything and not just LibTLS :P

File: Userland/Libraries/LibPthread/pthread.cpp [hunk 1]
```diff
@@ -557,12 +557,26 @@ int pthread_getname_np(pthread_t thread, char* buffer, size_t buffer_size)
 
 int pthread_setcancelstate([[maybe_unused]] int state, [[maybe_unused]] int* oldstate)
 {
-    TODO();
+    if (oldstate)
+        *oldstate = PTHREAD_CANCEL_DISABLE;
+    dbgln("FIXME: Implement pthread_setcancelstate({}, ...)", state);
+    if (state != PTHREAD_CANCEL_DISABLE) {
+        errno = EINVAL;
+        return -1;
```
[the remark this answers is not in the document]
Comment (gunnarbeutner) [reply Userland/Libraries/LibPthread/pthread.cpp]: Yup, gets me every time.

File: Userland/Libraries/LibPthread/pthread.cpp [hunk 2]
```diff
@@ -557,12 +557,22 @@ int pthread_getname_np(pthread_t thread, char* buffer, size_t buffer_size)
 
 int pthread_setcancelstate([[maybe_unused]] int state, [[maybe_unused]] int* oldstate)
```
Comment (linusg) [inline Userland/Libraries/LibPthread/pthread.cpp:558]: `[[maybe_unused]]` is no longer needed.

File: Userland/Libraries/LibPthread/pthread.cpp [hunk 3]
```diff
@@ -557,12 +557,22 @@ int pthread_getname_np(pthread_t thread, char* buffer, size_t buffer_size)
...
-    TODO();
+    if (oldstate)
+        *oldstate = PTHREAD_CANCEL_DISABLE;
+    dbgln("FIXME: Implement pthread_setcancelstate({}, ...)", state);
+    if (state != PTHREAD_CANCEL_DISABLE)
+        return EINVAL;
+    return 0;
 }
 
 int pthread_setcanceltype([[maybe_unused]] int type, [[maybe_unused]] int* oldtype)
```
Comment (linusg) [inline Userland/Libraries/LibPthread/pthread.cpp:568]: `[[maybe_unused]]` is no longer needed.

**Review by linusg — COMMENTED** [review]
I'd prefer `Fixes #1234.` over `fixes #1234`, but that's just a nitpick :^)

**Comment by gunnarbeutner** [comment]
> I'd prefer `Fixes #1234.` over `fixes #1234`, but that's just a nitpick :^)

Sure, will keep that in mind for next time. :)

**Pull request merged**